src/Microsoft.TestPlatform.Common/Utilities/FakesUtilities.cs

```diff
@@ -164,8 +186,30 @@ private static bool TryGetFakesDataCollectorConfigurator(out Func<IEnumerable<st
                 }
             }
 #endif
-            configurator = null;
-            return false;
+            return null;
+        }
+
+        private static Func<IDictionary<string, FrameworkVersion>, DataCollectorSettings> TryGetFakesNewDataCollectorConfigurator()
+        {
+            try
+            {
+                Assembly assembly = Assembly.Load(FakesConfiguratorAssembly);
+                var type = assembly?.GetType(ConfiguratorAssemblyQualifiedName, false);
+                var method = type?.GetMethod(ConfiguratorMethodName, new Type[] { typeof(IEnumerable<string>), typeof(FrameworkVersion) });
+                if (method != null)
+                {
+                    return (Func<IDictionary<string, FrameworkVersion>, DataCollectorSettings>)method.CreateDelegate(typeof(Func<IDictionary<string, FrameworkVersion>, DataCollectorSettings>));
+                }
+            }
+            catch (Exception ex)
+            {
+                if (EqtTrace.IsInfoEnabled)
+                {
+                    EqtTrace.Info("Failed to create newly implemented Fakes Configurator. Reason:{0} ", ex);
```

Comment:
Does this show up in test platform diagnostic logs? Its probably used in some places in this file already but eqttrace logs IIRC are enabled and viewed differently from how we enable TP diagnostic logs in general.